nix-in-docker/Dockerfile

```diff
@@ -0,0 +1,12 @@
+FROM nixos/nix
+
+VOLUME /nix
+VOLUME /root
+
+RUN nix-channel --add https://nixos.org/channels/nixpkgs-unstable nixpkgs
+RUN nix-channel --update
+RUN nix-env -i git
+
+COPY nix.conf /etc/nix/nix.conf
+
+CMD nix-shell
```

Comment:
Instead of installing git like that, then calling `nix-shell` as the CMD, we should instead do something like
```Dockerfile
COPY .. /mantis-ops
RUN nix-shell /mantis-ops --run "echo Finished"
RUN nix-env -f /mantis-ops -iA defaultNix.legacyPackages.x86_64-linux.git
```

(untested)

Reply:
That will copy the whole of current (build-time) mantis-ops into the image though, is that what we want? Git does seem special in that it is a prerequisite of running the nix commands, or am I missing something? I think it's OK if other nix dependencies are pulled in when running `nix-shell` for the first time in the container.

Reply:
Not sure how but this new image seems to be working much better and the custom image (+ installing git) isn't needed anymore. It's now just a `run` script (and separate `nix.conf` file.)